source/grid/grid_in.cc

```diff
@@ -411,10 +407,10 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
                         {
                           double id;
                           in >> id;
-                          if (set == "MaterialID")
+                          if (keyword == "MaterialID")
```

Comment:
AFAICT, these have different values. `set` is the next known data set name after `SCALARS` while we are still updating `keyword`.

Reply:
No, look at line 374-379 (old) above: We do `set = keyword` if we find something (and `continue` if we don't). I think this really is correct.

Reply:
Yes, but we change `keyword` in lines 389 and 395. I believe this is responsible for the failing tests.